pkg/common/connectionstore/connection_lookup.go

```diff
@@ -155,48 +165,100 @@ func (c *ConnectionLookup) GetConnectionRecordByNSThreadID(nsThreadID string) (*
 	if err != nil {
 		return nil, fmt.Errorf("get connectionID by namespaced threadID: %w", err)
 	}
-	// adding prefix for storing connection record
-	k := GetConnectionKeyPrefix()(string(connectionIDBytes))
 
-	connRecordBytes, err := c.transientStore.Get(k)
+	var rec ConnectionRecord
+
+	err = c.get(getConnectionKeyPrefix()(string(connectionIDBytes)), &rec, true)
 	if err != nil {
-		return nil, fmt.Errorf("get connection record by connectionID: %w", err)
+		return nil, fmt.Errorf("faild to get connection record by NS thread ID : %s, cause : %w", nsThreadID, err)
+	}
+
+	return &rec, nil
+}
+
+// GetInvitation finds and parses stored invitation to target type
+func (c *ConnectionLookup) GetInvitation(id string, target interface{}) error {
+	if id == "" {
+		return fmt.Errorf(errMsgInvalidKey)
 	}
 
-	return prepareConnectionRecord(connRecordBytes)
+	return c.get(getInvitationKeyPrefix()(id), target, false)
 }
 
-// Store returns handle to underlying permanent store
-func (c *ConnectionLookup) Store() storage.Store {
-	return c.store
+// GetEventData returns persisted event data for given connection ID
+// TODO connection event data shouldn't be transient [Issues #1029]
+func (c *ConnectionRecorder) GetEventData(connectionID string) ([]byte, error) {
+	if connectionID == "" {
+		return nil, fmt.Errorf(errMsgInvalidKey)
+	}
+
+	return c.transientStore.Get(getEventDataKeyPrefix()(connectionID))
 }
 
-// TransientStore returns handle to underlying transient store
-func (c *ConnectionLookup) TransientStore() storage.Store {
-	return c.transientStore
+func (c *ConnectionLookup) get(key string, target interface{}, temp bool) error {
```

Comment:
I also don't quite understand the purpose of this function signature.

- temp?
- empty interface?

Reply:
I also didn't quite understand from the func name (and lack of func comment), that this method both retrieves and unmarshals a record.

Reply:
this is not exported actually, I just used it to switch between stores and to test that part separately to improve code coverage. Will fix it.

Reply:
Changed it as suggested. 
getAndUnmarshal & marshalAndSave